### PR TITLE
simplify query audit logging

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Simplify and improve audit logging.
Previously, even if audit logging is turned off, the audit log messages are always generated, only to be discarded quickly. This is inefficient and not necessary. This PR rectifies the situation by only creating the audit log messages when actually required.
In addition, the audit logging code has been slightly simplified, removing the previous if..then..else cases from all events that made a distinction between internal and non-internal operations.

Audit logging for AQL queries has also been improved to include the query runtime and the "exit code" (which is 0 for "no error", and > 0 to indicate the corresponding ArangoDB error).

Enterprise companion PR: https://github.com/arangodb/enterprise/pull/609

- [ ] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [x] :fire: Performance improvement
- [x] :hammer: Refactoring/simplification
- [x] :book: CHANGELOG entry made

#### Backports:

- [x] No backports required

### Testing & Verification

- [x] The behavior in this PR was *manually tested*
- [x] This change is already covered by existing tests, such as *scripts/unittest audit_client audit_server*.

http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13297/

### CODE REVIEW

**Review by fceller — COMMENTED** [review]
LGTM

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13309/

File: 3rdParty/velocypack/include/velocypack/Slice.h [hunk 1]
```diff
@@ -52,6 +53,10 @@
 
 namespace arangodb {
 namespace velocypack {
+class Sink;
```
Comment (jsteemann) [inline 3rdParty/velocypack/include/velocypack/Slice.h:56]:
```suggestion
struct Sink;
```

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13315/

**Pull request merged**